notebook/services/contents/filemanager.py

```diff
@@ -438,7 +438,10 @@ def save(self, model, path=''):
         try:
             if model['type'] == 'notebook':
                 nb = nbformat.from_dict(model['content'])
-                self.check_and_sign(nb, path)
+                # We need this check in case the notebook is 
+                # trusted before save
+                if not self.notary.check_signature(nb):
+                    self.check_and_sign(nb, path)
```

Comment:
check_signature shouldn't be called before `check_and_sign`, which consumes trust metadata. I think `.save` probably doesn't need to change at all.